File: Selection Sort/selection_sort.py
import numpy as np
from random import shuffle


def selection_sort(array):
    for i in range(len(array)):
        
        smallest_value_index = i
        
        # Search for the smallest value index
        for j in  range(i, len(array)):
            if array[j] < array[smallest_value_index]:
                smallest_value_index = j

        array[i], array[smallest_value_index] = array[smallest_value_index], array[i]

    return array


# Buat array berisi 0 sampai 999
random_array = np.arange(1000)

# Acak isi array
shuffle(random_array)

print(random_array)

print(selection_sort(random_array))

File: Selection Sort/selection-sort.js
function selectionSort(array) {
	var temp;

	for (var i = 0; i < array.length; i++) {
		var smallestIndex = i;
		for (var j = i; j < array.length; j++) {
			if (array[smallestIndex] > array[j]) {
				smallestIndex = j;
			}
		}

		// swap current i & smallest index
		temp = array[smallestIndex];
		array[smallestIndex] = array[i];
		array[i] = temp;
	}

	return array;
}

let array = [];

dataLength = 100

for (var i = 0; i < dataLength; i++) {
	// Get random number from 1 to dataLength
	var randomNumber = i;
	array.push(randomNumber);
}

// shuffle array
array = array.sort(() => Math.random() - 0.5)

console.log(array);
var startTime = new Date();

sortedArray = selectionSort(array);

endTime = new Date();

var timeDiff = endTime - startTime; //in ms
console.log(timeDiff + "ms sorting time.");
// strip the ms
timeDiff /= 1000;

// get seconds 
var seconds = Math.round(timeDiff);
console.log(seconds + " seconds elapsed.")

console.log(sortedArray);
